kernel/aster-nix/src/fs/path/mount.rs

```diff
@@ -97,6 +97,89 @@ impl MountNode {
         Ok(child_mount)
     }
 
+    /// Clone a mount node with the an root Dentry_.
+    /// The new mount node will have the same fs as the original one and
+    /// have no parent and children. We should set the parent and children manually.
+    pub fn clone_mount_node(&self, root_dentry: Arc<Dentry_>) -> Arc<Self> {
+        Arc::new_cyclic(|weak_self| Self {
+            root_dentry,
+            mountpoint_dentry: RwLock::new(None),
+            parent: RwLock::new(None),
+            children: Mutex::new(BTreeMap::new()),
+            fs: self.fs.clone(),
+            this: weak_self.clone(),
+        })
+    }
+
+    /// Copies a mount tree starting from the specified root `Dentry_`.
+    /// The new mount tree will replicate the structure of the original tree.
+    /// The new tree is a separate entity rooted at the given `Dentry_`,
+    /// and the original tree remains unchanged.
+    pub fn copy_mount_node_tree(&self, root_dentry: Arc<Dentry_>) -> Arc<Self> {
```

Comment:
This should be a private method.

Reply:
Thank you for your review and the suggestion to make `copy_mount_node_tree` a private method. However, there are use cases where `clone_mount_node_tree` could be utilized within Dentry, such as in the `Dentry::bind_mount_to`:
```rust
pub fn bind_mount_to(&self, dst_dentry: &Arc<Self>, recursive: bool) -> Result<()> {
    let src_mount = self
        .mount_node
        .clone_mount_node_tree(&self.inner, recursive);
    src_mount.graft_mount_node_tree(dst_dentry)?;
    Ok(())
}
```
Considering this, I believe it would be more appropriate to use pub(super) for `clone_mount_node_tree` to indicate that it may be used by `Dentry`. 

Furthermore, in the context of the mount namespace, there may be operations involving clone_mount_node_tree where the process's root and current working directory are moved to the new mount node tree.